app/controllers/microposts_controller.rb

```diff
@@ -4,7 +4,7 @@ class MicropostsController < ApplicationController
   # GET /microposts
   # GET /microposts.json
   def index
-    @microposts = Micropost.search(params[:search]).page(params[:page])
+    @microposts = Micropost.category(params[:category]).search(params[:search]).page(params[:page])
```

Comment:
It's better to add `order` otherwise database won't sort the records. Sometime results will be random.

Reply:
oh right thank you, I forgot I been using sqlite 😆